requirements/dev.txt

```diff
@@ -10,7 +10,7 @@ mypy==1.8.0
 nox
 pre-commit
 pyflakes
-pytest
+pytest<8.4  # Pin to <8.4 due to removal of RaisesContext from private API
```

Comment:
how unfortunate

Reply:
Yeah... I spent a while looking into supporting the new `RaisesExc` simultaneously and it is a big mess. I think a separate PR will be needed to switch to `pytest>=8.4` and switch to solely supporting `RaisesExc`. Refactoring the tests that use it to use neither seems pretty non-trivial